modules/http/src/main/java/org/mule/module/http/internal/request/HttpRequesterRequestBuilder.java

```diff
@@ -39,12 +43,16 @@ public String replaceUriParams(String path, MuleEvent event)
 
             if (uriParamValue == null)
             {
-                throw new NullPointerException(String.format("Expression {%s} evaluated to null.", uriParamName));
+                throw new NullPointerException(format("Expression {%s} evaluated to null.", uriParamName));
             }
 
-            path = path.replaceAll(String.format("\\{%s\\}", uriParamName), uriParamValue);
+            String value = quoteReplacement(uriParamValue);
```

Comment:
Why was this necessary?

Reply:
It's necessary to solve the dollar sign problem, replaceAll method documentation: "Note that backslashes and dollar signs in the replacement string may cause the results to be different than if it were being treated as a literal replacement string. Dollar signs may be treated as references to captured subsequences as described above, and backslashes are used to escape literal characters in the replacement string."

Reply:
inline it